### PR TITLE
boards/lobaro-lorabox: Add support for Lobaro LoraBox board

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This adds support for the [Lobaro LoraBox](https://www.lobaro.com/portfolio/wireless-mbus-zu-lora-bridge-adapter-repeater) board. It is based on the stm32l151cb-a MCU.
Lobaro LoraBox

### Testing procedure
With the provided UART-USB bridge connected to the "Config" port on the board run:
```
 BOARD=lobaro-lorabox make flash term -C examples/hello-world
```

### Issues/PRs references
Depends on  ~#9802~

### CODE REVIEW

File: boards/lobaro-lorabox/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# configure the serial terminal
+PORT_LINUX ?= /dev/ttyUSB0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+FLASHER = $(RIOTTOOLS)/stm32loader/stm32loader.py
+HEXFILE = $(BINFILE)
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply boards/lobaro-lorabox/Makefile.include]: Yes you are right, I will just move this to the ```FFLAGS```.

File: boards/lobaro-lorabox/Makefile.features [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_gpio
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_rtc
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+
+# The board MPU family (used for grouping by the CI system)
+FEATURES_MCU_GROUP = cortex_m3_2
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply boards/lobaro-lorabox/Makefile.features]: I will remove it.

[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply boards/lobaro-lorabox/Makefile.features]: I will remove it.

File: boards/lobaro-lorabox/Makefile.include (hunk 1 above)
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply boards/lobaro-lorabox/Makefile.include]: Yes, you are right, I will just move it to ```FFLAGS```

[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply boards/lobaro-lorabox/Makefile.include]: Actually I just realized that if I don't assign ```BINFILE``` to ```HEXFILE``` the .bin file is not generated, so I will have to keep it this way.

**Comment by MrKevinWeiss** [comment]
@kYc0o I tested the board, I will leave it up to you to make the big green button!

File: boards/lobaro-lorabox/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+ */
+
+/**
+ * @defgroup    boards_lorabox Lobaro lorabox
+ * @ingroup     boards_lorabox
+ * @brief       Support for the Lobaro lorabox with stm32l151cb
+ * @{
+ *
+ * @file
+ * @brief       Peripheral MCU configuration for theLobaro lorabox board
```
Comment (kYc0o) [inline boards/lobaro-lorabox/include/periph_conf.h]: Small typo, please add the required space.

File: boards/lobaro-lorabox/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+/*
+ * 0: no external low speed crystal available,
+ * 1: external crystal available (always 32.768kHz)
+ *
+ * LSE might not be available by default in early (C-01) Nucleo boards.
+ * For newer revisions, an LSE crystal is present and CLOCK_LSE can be set to 1
+ * if one wants to use it.
+ */
+#ifndef CLOCK_LSE
+#define CLOCK_LSE           (1)
```
Comment (kYc0o) [inline boards/lobaro-lorabox/include/periph_conf.h]: Have you tested this? You should also remove/modify this comment since it comes from a Nucleo board.

**Comment by kYc0o** [comment]
As for the commits, I'd suggest to change them as following:

  - First commit for the CPU changes which add the cpu model to the `#ifdef` statements.
  - Second commit with all the board related stuff
  - Third commit adding the flashing tools

**Comment by kYc0o** [comment]
> I tested:
>  - RTC
>  - CLOCK_LSE = 1 and CLOCK_LSE = 0

Did they work correctly?

**Comment by leandrolanzieri** [comment]
> > I tested:
> > 
> > * RTC
> > * CLOCK_LSE = 1 and CLOCK_LSE = 0
> 
> Did they work correctly?

They did work.
Now I am introducing some other changes on the CPU because of the failing tests.

> As for the commits, I'd suggest to change them as following:
> 
> * First commit for the CPU changes which add the cpu model to the `#ifdef` statements.
> * Second commit with all the board related stuff
> * Third commit adding the flashing tools

I will try to do this but I have some mixed modifications in some commits.

**Comment by leandrolanzieri** [comment]
> As for the commits, I'd suggest to change them as following:
> 
> * First commit for the CPU changes which add the cpu model to the `#ifdef` statements.
> * Second commit with all the board related stuff
> * Third commit adding the flashing tools

I modified the commits so they follow this logic.

**Comment by kYc0o** [comment]
Let's go then.

**Pull request merged**